File: app11_swift_ios/SwiftGuardUI/Sources/SwiftGuardUI/Views/DigitalHarmsView.swift
import SwiftData
import SwiftUI
import SwiftGuardData

/// US-19: dedicated, not routed through `CardSuitView` — this deck must
/// NEVER render a severity badge (FR-19.2), and `CardKind.severity` is
/// structurally `nil` for every card here (D-03), so there is no field to
/// print even by copy-paste mistake.
public struct DigitalHarmsView: View {
    @Environment(\.modelContext) private var modelContext
    @Environment(LocalizationManager.self) private var localizationManager
    @State private var viewModel: CardSuitViewModel?

    private static let suitOrder = ["SCO", "ARC", "AGE", "TRU", "POR"]

    public init() {}

    public var body: some View {
        List {
            Section {
                Text("Ta talia nie jest listą podatności technicznych z poziomem severity — modeluje harmy projektowe (wykluczenie cyfrowe, nieprzejrzyste projektowanie) w usługach publicznych, mapowane na OWASP A04:2021 Insecure Design.")
                    .font(.caption)
            }

            ForEach(Self.suitOrder, id: \.self) { suit in
                Section(suit) {
                    ForEach(cards(forSuit: suit)) { card in
                        VStack(alignment: .leading, spacing: 4) {
                            HStack {
                                Text(card.cardId).font(.caption.bold())
                                Text("harm projektowy")
                                    .font(.caption2.bold())
                                    .padding(.horizontal, 4)
                                    .background(.purple.opacity(0.2))
                                    .foregroundStyle(.purple)
                                    .clipShape(Capsule())
                            }
                            Text(card.localizedDescription(localizationManager.currentLocale)).font(.subheadline)
                            if !card.owaspRefs.isEmpty {
                                Text(card.owaspRefs.joined(separator: ", ")).font(.caption2).foregroundStyle(.secondary)
                            }
                        }
                    }
                }
            }
        }
        .navigationTitle("Digital-by-Default Harms")
        .task {
            if viewModel == nil {
                viewModel = CardSuitViewModel(modelContext: modelContext)
            }
            viewModel?.loadEdition("dbd")
        }
    }

    private func cards(forSuit suit: String) -> [CornucopiaCard] {
        (viewModel?.cards ?? []).filter { $0.suitCode == suit }
    }
}

File: app13_ruby_FastApi/frontend/src/views/digital-harms-view.js
import { api } from "../api-client.js";
import { getLocale } from "../i18n.js";

// US-19/D-03: this view structurally cannot render a severity badge — the
// API's `severity` field is always `null` for `edition=dbd` cards (enforced
// by the database CHECK constraint, PLAN.md §4 D-03), and this template
// simply never references `card.severity` at all, unlike card-suit-view.js.
export async function render(container) {
  container.innerHTML = "<p>Loading…</p>";
  const cards = await api.cards({ edition: "dbd" });
  const locale = getLocale();

  container.innerHTML = `
    <h1>Digital-by-Default Harms</h1>
    <p class="muted">
      Ta talia nie jest listą podatności technicznych z poziomem severity — modeluje harmy
      projektowe (wykluczenie cyfrowe, nieprzejrzyste projektowanie) w usługach publicznych,
      mapowane na OWASP A04:2021 Insecure Design.
    </p>
    <ul class="card-list">
      ${cards
        .map((card) => {
          const description = locale === "pl" && card.descriptionPl ? card.descriptionPl : card.descriptionEn;
          return `
            <li data-testid="card-row-${card.cardId}">
              <strong>${card.cardId}</strong> <span class="badge design-harm">harm projektowy</span>
              <p>${description}</p>
              ${card.owaspRefs.length ? `<p class="muted">${card.owaspRefs.join(", ")}</p>` : ""}
            </li>`;
        })
        .join("")}
    </ul>`;
}
